online-shopping-website/src/Components/ProductDetails.js

```diff
@@ -0,0 +1,128 @@
+        super(props);
+        this.state = {
+            quantity: 1,
+            show: true,
+            max: 5,
+            min: 0
+        };
+    }
+
+    IncrementItem = () => {
+        if (this.state.quantity != 10) {
+            this.setState({
+                quantity: this.state.quantity + 1
+            });
+        }
+    }
+    DecreaseItem = () => {
+        if (this.state.quantity != 1) {
+            this.setState({quantity: this.state.quantity - 1});
+        }
+    }
+
+    UpdateValue = (e) => {
+        this.setState({quantity: e.target.value});
+    }
+
+    get quantity() {
+        return this.state.quantity;
+    }
+
+    render() {
+        return (
+            <div className="ProductDetails-QuantityButtonsContainer">
+                <h3>Quantity</h3>
+                <Stack className="ProductDetails-QuantityButtonsStack" direction="row" spacing={1}>
+                    <Button className="ProductDetails-QuantityButtons" variant="contained"
+                            disabled={this.state.quantity == 10}
+                            onClick={this.IncrementItem}>
+                        <AddIcon/>
+                    </Button>
+                    <input className="inputne" value={this.state.quantity} onChange={this.UpdateValue}/>
+                    <Button className="ProductDetails-QuantityButtons" variant="contained"
+                            disabled={this.state.quantity == 1}
+                            onClick={this.DecreaseItem}>
+                        <RemoveIcon/>
+                    </Button>
+                </Stack>
+                <h5 className="ProductDetails-ProductLimitText">Limit of 10 items per product in cart.</h5>
+                <Button className="ProductDetails-CartButton" variant="contained" endIcon={<AddShoppingCartIcon/>}>
+                    Add to cart
+                </Button>
+            </div>
+        );
+    }
+}
+
+export const ProductDetails = () => {
+    const productName = useParams();
+
+    const selectedProduct = {
+        brand: "",
+        description: "",
+        id: null,
+        image: "",
+        name: "",
+        price: null,
+        seller: ""
+    }
+
+    Products.products.map(product => {
+        if (productName.productName === product.name) {
+            selectedProduct.brand = product.brand;
+            selectedProduct.description = product.description;
+            selectedProduct.id = product.id;
+            selectedProduct.image = product.image;
+            selectedProduct.name = product.name;
+            selectedProduct.price = product.price.toFixed(2);
+            selectedProduct.seller = product.seller;
+        }
+    });
+
+    return (
+        <Grid container className="ProductDetails-Container">
+            <Grid item xs={12} sm={9} lg={8}>
+                <h1>{selectedProduct.name}</h1>
+                <img className="ProductDetails-Image" src={selectedProduct.image} alt={selectedProduct.name}/>
```

Comment:
The image tends to press up against the right side bar that holds the price and buttons. Maybe add some padding to the right of the image or make all product images a static size to fix this? As long as they're consistent it should be good!